joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller_impl.h

```diff
@@ -138,6 +138,13 @@ starting(const ros::Time& time)
   time_data.uptime = ros::Time(0.0);
   time_data_.initRT(time_data);
 
+  // Initialize the desired_state with the current state on startup
```

Comment:
This does not change the behavior, it gets overwritten in `update` anyway.

Reply:
exactly: It gets overwritten on the next update(). But `starting` calls `setHoldPosition` three lines later and reads the desired_state (if `stop_trajectory_duration_ != 0`), so we have to initialize it beforehand.

Reply:
Yes, I have just written this for clarification/justfication during the review :)